### ISSUE TITLE
single saga exception message being processed by multiple instances during retry and delay

### ISSUE THREAD

**Issue by jsurana** [Particular/NServiceBus#6600]
Hi,
 I am using NServicebus Core with File version 7.7.4.0. I am having two instances for one handler . I have retry and delayed mechanism as well. So when any exception occurs, both the instances are retrying. How can I avoid this? I want if any exception occurs, only single instance should retry at a time. I want to keep two instance mandatorily. Is it possible?

**Comment by mauroservienti** [comment]
I'm not sure I understand the whole scenario, @jsurana.

What should be happening is that the first batch of retries (first level) is performed by the instance that receives the message. If delayed retries kick in when the message is retried will be handled by one of the available instances, not necessarily the first that initially handled the message.

What should not happen is that the same message is handled simultaneously by more than one instance. is that what you're observing?

**Comment by jsurana** [comment]
@mauroservienti We have two instances which are processing a message at the same time when exception occurs. Please find below screenshot for more details:

![image](https://user-images.githubusercontent.com/31936315/203970557-f163d8dd-5b0b-4e72-9ce6-01f18bf7bc5a.png)

**Comment by mauroservienti** [comment]
I need more details to better understand the involved endpoints configuration.

To avoid publicly disclosing sensitive details, can you please send at "support (at) particular.net" the following information?

- NServiceBus version 
- transport type and version 
- the sender endpoint configuration code
- the sender code that sends the failing message
- the receiving endpoint configuration 
- the receiving message handling code

Thanks.

**Comment by jsurana** [comment]
@mauroservienti do you mean to say, if message is timeout and there is load balancing available with two instances/nodes then both instances will process same message at the same time?

**Comment by mauroservienti** [comment]
No, the opposite. Only one instance should process the message.

**Comment by jsurana** [comment]
> I need more details to better understand the involved endpoints configuration.
> 
> To avoid publicly disclosing sensitive details, can you please send at "support (at) particular.net" the following information?
> 
> * NServiceBus version
> * transport type and version
> * the sender endpoint configuration code
> * the sender code that sends the failing message
> * the receiving endpoint configuration
> * the receiving message handling code
> 
> Thanks.

@mauroservienti I have shared the details over the email. Can you please check email.

**Comment by mauroservienti** [comment]
@jsurana is there any update on this issue?
